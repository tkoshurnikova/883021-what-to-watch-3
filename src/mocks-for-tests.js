export const FilmDetails = {
  title: `Whispering Mist`,
  genre: `Comedy`,
  year: 1995,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};

export const films = [
  {
    "title": `The No Flowers`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `Rings of Cloud`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `The Sword's Memory`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `The Voyagers of the Words`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `Shores in the Heart`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `Missing Illusion`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `The Unwilling Man`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  },
  {
    "title": `Teacher in the Hunter`,
    "image": `http://placehold.it/280x175`,
    "description": `qwerty`,
    "rating": 1.0,
    "numberOfVotes": 1982,
    "director": `Taika Di Caprio`,
    "actors": [`Ashley Cooper`, `Ramona Flowers`],
    "genre": `Dramas`,
    "year": 1900,
    "preview": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "video": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    "background_color": `#ffffff`,
    "background_image": `some image`,
    "poster_image": `some image`
  }
];
