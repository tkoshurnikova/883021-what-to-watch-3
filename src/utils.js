export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const convertSecondsToHours = (time) => {
  let hours = Math.floor(time / 60 / 60);
  let minutes = Math.floor(time / 60) - (hours * 60);
  let seconds = time % 60;
  let formatted = hours.toString().padStart(2, `0`) + `:` + minutes.toString().padStart(2, `0`) + `:` + seconds.toString().padStart(2, `0`);
  return formatted;
};

export const getFilmByID = (films, id) => films.find((film) => film.id === Number(id));

export const noop = () => {};
