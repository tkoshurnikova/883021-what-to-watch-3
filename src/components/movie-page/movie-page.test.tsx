import * as React from "react";
import * as renderer from "react-test-renderer";
import MoviePage from "./movie-page";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {films} from "../../mocks-for-tests";
import {Router} from "react-router-dom";
import history from "../../history";
import {noop} from "../../utils";

const mockStore = configureMockStore([]);

it(`Render MoviePage`, () => {
  const store = mockStore({
    APP: {
      genre: ``,
      cardsToShow: 8
    },
    DATA: {
      films,
      favoriteFilms: []
    },
    USER: {
      isAuthorized: false
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <MoviePage
              film={films[0]}
              onActiveItemChange={noop}
              activeItem={``}
            />
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
