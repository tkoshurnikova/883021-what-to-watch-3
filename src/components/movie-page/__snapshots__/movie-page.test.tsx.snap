// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MoviePage 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The No Flowers"
          src="some image"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The No Flowers
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Dramas
            </span>
            <span
              className="movie-card__year"
            >
              1900
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height={19}
                viewBox="0 0 19 19"
                width={19}
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height={20}
                viewBox="0 0 19 20"
                width={19}
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/films/1/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The No Flowers"
            height={327}
            src="some image"
            width={218}
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item movie-nav__item"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item movie-nav__item"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <img
                alt="Rings of Cloud"
                height="175"
                src="http://placehold.it/280x175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Rings of Cloud
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <img
                alt="The Sword's Memory"
                height="175"
                src="http://placehold.it/280x175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              The Sword's Memory
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <img
                alt="The Voyagers of the Words"
                height="175"
                src="http://placehold.it/280x175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              The Voyagers of the Words
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <a
            href="/films/5"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <img
                alt="Shores in the Heart"
                height="175"
                src="http://placehold.it/280x175"
                width="280"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              Shores in the Heart
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
